lib/fencing.py.py

```diff
@@ -1295,64 +1301,130 @@ def _set_default_values(options):
 	return options
 
 # stop = True/False : exit fence agent when problem is encountered
-def _validate_input(options, stop = True):
+def _validate_input(options, stop = True, output_format = 'text'):
 	device_opt = options["device_opt"]
 	valid_input = True
+	errors = []
 
 	if "--username" not in options and \
 			device_opt.count("login") and (device_opt.count("no_login") == 0):
-		valid_input = False
-		fail_usage("Failed: You have to set login name", stop)
+		errors.append({
+			'fields': ['login'],
+			'text': 'You have to enter login name',
+			'text_cli': 'You have to enter login name (-l)',
+			'error_type': 'REQUIRE',
+			'error': True,
+		})
 
 	if device_opt.count("ipaddr") and "--ip" not in options and "--managed" not in options and "--target" not in options:
-		valid_input = False
-		fail_usage("Failed: You have to enter fence address", stop)
+		errors.append({
+			'fields': ['ipaddr', 'managed', 'target'],
+			'text': 'You have to enter fence address',
+			'text_cli': 'You have to enter fence address (-a)',
+			'error_type': 'REQUIRE',
```

Comment:
Is this one-or-more, vs one-of being one or the other?

Reply:
one-or-more; which name do you suggest? ONE-OR-MORE?